File: FrontEnd/lib/domain/interfaces/usecases/user/i_delete_user_account_use_case.dart
/// @file i_delete_user_account_use_case.dart
/// @description Interfaz: Elimina la cuenta del usuario (soft-delete).
/// @module User
/// @layer Domain
library;
abstract interface class IDeleteUserAccountUseCase {
  /// Elimina la cuenta del usuario (soft-delete).
  ///
  /// [param] password        — contraseña actual para confirmar la eliminación.
  /// [param] preserveContent — si true, las publicaciones y comentarios del usuario
  ///                           permanecen (anónimos); si false (default), se eliminan.
  Future<void> execute(String password, {bool preserveContent = false});
}

File: BackEnd/src/domain/interfaces/usecases/user/IDeleteUserAccountUseCase.ts
/**
 * @file IDeleteUserAccountUseCase.ts
 * @description Interfaz del caso de uso para eliminar la cuenta del usuario (soft-delete).
 * @module User
 * @layer Domain
 */
export interface IDeleteUserAccountUseCase {
  /**
   * @param userId          — ID del usuario autenticado.
   * @param password        — Contraseña actual para confirmar la eliminación.
   * @param preserveContent — Si true, las publicaciones y comentarios del usuario
   *                          se mantienen (anónimos). Si false (default), se eliminan.
   */
  execute(userId: string, password: string, preserveContent?: boolean): Promise<void>;
}
